src/extract.js

```diff
@@ -5,7 +5,9 @@ async function run() {
   try {
     const image = core.getInput('image');
     const path = core.getInput('path');
-    const destination = `.extracted-${Date.now()}`;
+    let destination = core.getInput('destination');
+
+    if (!destination) destination = `.extracted-${Date.now()}`;
```

Comment:
The default behaviour of the action is for destination path to be the generated path, and you are adding the option to specify the path. If we have the code default to the extracted path, and then optionally overwrite it when a `destination` is specified, that'll best mirror the behaviour, e.g:

```suggestion
    let destination = `.extracted-${Date.now()}`;
    
    if (core.getInput('destination')) {
      let destination = core.getInput('destination')
    }
```

Alternatively, this could be shortened using a ternary, e.g:

```suggestion
    const destination = core.getInput('destination') ? core.getInput('destination') : `.extracted-${Date.now()}`;
```

Or made even simpler by taking advantage of `core.getInput()`'s behaviour of returning an empty string which can be used with an [OR](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_OR), e.g:

```suggestion
    const destination = core.getInput('destination') || `.extracted-${Date.now()}`;
```

Personally, I'd probably go with the last option but I'm happy for you to choose whichever you think is best.